### PR TITLE
Update: deprecate id-blacklist rule

### PR DESCRIPTION
**Pull request by dimitropoulos**
#### Prerequisites checklist

- [x] I have read the [contributing guidelines](https://github.com/eslint/eslint/blob/master/CONTRIBUTING.md).

#### What is the purpose of this pull request? (put an "X" next to an item)

[x] Documentation update
[x] Bug fix ([template](https://raw.githubusercontent.com/eslint/eslint/master/templates/bug-report.md))
[ ] New rule ([template](https://raw.githubusercontent.com/eslint/eslint/master/templates/rule-proposal.md))
[x] Changes an existing rule ([template](https://raw.githubusercontent.com/eslint/eslint/master/templates/rule-change-proposal.md))
[ ] Add autofixing to a rule
[ ] Add a CLI option
[ ] Add something to the core
[ ] Other, please explain:

#### What changes did you make? (Give an overview)

I noticed that in the haste to rename `id-blacklist` to `id-denylist`, `id-blacklist` was not deprecated.  This PR aims to mark `id-blacklist` as deprecated and document this change.

Regarding documentation, note that some tooling of tooling (well, tooling I use, anyway) expects the rule to have documentation matching the name of the rule.

#### Is there anything you'd like reviewers to focus on?

Just to make sure this was done in the desired/appropriate manner.

### CODE REVIEW

**Comment by kaicataldo** [comment]
The decision not to deprecate the rule was a conscious one. We thought that creating a URL redirect on the website would be enough. I'm not against marking the rule as deprecated if the direction we took is breaking things, but do you mind sharing a concrete example of what's breaking?

**Comment by dimitropoulos** [comment]
sure: for example I use https://github.com/sarbbottam/eslint-find-rules for some configs I maintain (for example, one is [eslint-config-intense](https://github.com/dimitropoulos/eslint-config-intense/blob/master/index.js)) and it will ask me to define rules for both `id-blacklist` and `id-denylist` since they are not both deprecated.

**Comment by kaicataldo** [comment]
Thanks for sharing that. @eslint/eslint-tsc Thoughts on how we should proceed here?

**Comment by nzakas** [comment]
I think this makes sense. We were operating on the theory that there wouldn't be problems with making this switch silently, but that now appears not to be the case. Deprecation should solve the problem.

**Comment by kaicataldo** [comment]
Sounds good to me. @btmills @mysticatea @mdjermanovic thoughts?

**Comment by mysticatea** [comment]
Deprecation sounds good because it looks confusable for tools if the same rule exists as two different names and both are not deprecated.

I have a small concern about the deprecated rule imports the living rule. We have made the code of deprecated rules frozen until this. Therefore, I feel odd a bit if we will continue to update the new deprecated rule. But it's not a strong feeling because this is rename.

**Comment by dimitropoulos** [comment]
@mysticatea that's fair.  I was trying to import `id-denylist`'s code into `id-blacklist` to reduce duplication, but I'd be happy to copy-pasta if that serves the use-case better for freezing a deprecated rule.

**Comment by mdjermanovic** [comment]
I also think it would be consistent to copy & paste the code, and thus freeze the deprecated rule.

**Comment by btmills** [comment]
Looks like y'all have this covered. :+1: to deprecation since it seems like the best path forward, and while I respect the DRYness of the current implementation, :+1: to copy & paste since we have previous deprecations as precedent.

**Comment by dimitropoulos** [comment]
Thanks all: I'll definitely jump on that soon.  While I do so, may I suggest that we document this rule about deprecated rules being frozen in developer docs somewhere.  I think it's a great policy to have and it seems like it's worth making official.

I'd be happy to make a issue+PR if so (we can discuss where it should live in the issue maybe?)

File: docs/rules/id-blacklist.md [hunk 1]
```diff
@@ -0,0 +1,3 @@
+# disallow specified identifiers (id-blacklist)
+
+This rule has been deprecated.  Please see the [`id-denylist`]('./id-blacklist') rule.
```
Comment (mdjermanovic) [inline docs/rules/id-blacklist.md:3]:
```suggestion
This rule was **deprecated** in ESLint v7.5.0 and replaced by the [id-denylist](id-denylist.md) rule.
```

**Comment by dimitropoulos** [comment]
ok! looks like this is all set (as far as I'm aware)!

File: lib/rules/id-blacklist.js [hunk 2]
```diff
@@ -0,0 +1,232 @@
...
+    );
+}
+
+//------------------------------------------------------------------------------
+// Rule Definition
+//------------------------------------------------------------------------------
+
+module.exports = {
+    meta: {
+        deprecated: true,
```
Comment (mdjermanovic) [inline lib/rules/id-blacklist.js:114]: We can also add `replacedBy: ["id-denylist"]`, it's used for the [Deprecated](https://eslint.org/docs/rules/#deprecated) table, and maybe by some tools.

**Comment by mdjermanovic** [comment]
> Thanks all: I'll definitely jump on that soon. While I do so, may I suggest that we document this rule about deprecated rules being frozen in developer docs somewhere. I think it's a great policy to have and it seems like it's worth making official.

I think the policy is already documented here:

https://eslint.org/docs/user-guide/rule-deprecation

Maybe it isn't explicit enough that the behavior of the rule will never change?

Hm, actually if it happens that a deprecated rule uses some shared helpers like `astUtils`, its behavior can be changed indirectly.

**Comment by mdjermanovic** [comment]
It seems there is a consensus to do this, so I'm marking this PR as accepted.

File: lib/rules/id-blacklist.js (hunk 2 above)
Comment (dimitropoulos) [reply lib/rules/id-blacklist.js:114]: cool! didn't know about that!  when is eslint being rewritten in TypeScript, again? :laughing: just joking.

**Comment by kaicataldo** [comment]
Thanks for contributing!

**Pull request merged**